## BUSINESS UNDERSTANDING

In the modern 21st Century, communication has been advanced to data transmission through text, voice,audio or video across the globe. This has lead to the emergence of companies to support the process

## DATA UNDERSTANDING

In [39]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [40]:
# opening the data
df = pd.read_csv('Data/bigml_59c28831336c6604c800002a.csv')
df.head()

state  account length  area code phone number international plan   
0    KS             128        415     382-4657                 no  \
1    OH             107        415     371-7191                 no   
2    NJ             137        415     358-1921                 no   
3    OH              84        408     375-9999                yes   
4    OK              75        415     330-6626                yes   

  voice mail plan  number vmail messages  total day minutes  total day calls   
0             yes                     25              265.1              110  \
1             yes                     26              161.6              123   
2              no                      0              243.4              114   
3              no                      0              299.4               71   
4              no                      0              166.7              113   

   total day charge  ...  total eve calls  total eve charge   
0             45.07  ...               99             16.78  \
1             27.47  ...              103             16.62   
2             41.38  ...              110             10.30   
3             50.90  ...               88              5.26   
4             28.34  ...              122             12.61   

   total night minutes  total night calls  total night charge   
0                244.7                 91               11.01  \
1                254.4                103               11.45   
2                162.6                104                7.32   
3                196.9                 89                8.86   
4                186.9                121                8.41   

   total intl minutes  total intl calls  total intl charge   
0                10.0                 3               2.70  \
1                13.7                 3               3.70   
2                12.2                 5               3.29   
3                 6.6                 7               1.78   
4                10.1                 3               2.73   

   customer service calls  churn  
0                       1  False  
1                       1  False  
2                       0  False  
3                       2  False  
4                       3  False  

[5 rows x 21 columns]

In [41]:
df.tail()

state  account length  area code phone number international plan   
3328    AZ             192        415     414-4276                 no  \
3329    WV              68        415     370-3271                 no   
3330    RI              28        510     328-8230                 no   
3331    CT             184        510     364-6381                yes   
3332    TN              74        415     400-4344                 no   

     voice mail plan  number vmail messages  total day minutes   
3328             yes                     36              156.2  \
3329              no                      0              231.1   
3330              no                      0              180.8   
3331              no                      0              213.8   
3332             yes                     25              234.4   

      total day calls  total day charge  ...  total eve calls   
3328               77             26.55  ...              126  \
3329               57             39.29  ...               55   
3330              109             30.74  ...               58   
3331              105             36.35  ...               84   
3332              113             39.85  ...               82   

      total eve charge  total night minutes  total night calls   
3328             18.32                279.1                 83  \
3329             13.04                191.3                123   
3330             24.55                191.9                 91   
3331             13.57                139.2                137   
3332             22.60                241.4                 77   

      total night charge  total intl minutes  total intl calls   
3328               12.56                 9.9                 6  \
3329                8.61                 9.6                 4   
3330                8.64                14.1                 6   
3331                6.26                 5.0                10   
3332               10.86                13.7                 4   

      total intl charge  customer service calls  churn  
3328               2.67                       2  False  
3329               2.59                       3  False  
3330               3.81                       2  False  
3331               1.35                       2  False  
3332               3.70                       0  False  

[5 rows x 21 columns]

In [42]:
df.shape

(3333, 21)

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3333 entries, 0 to 3332
Data columns (total 21 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   state                   3333 non-null   object 
 1   account length          3333 non-null   int64  
 2   area code               3333 non-null   int64  
 3   phone number            3333 non-null   object 
 4   international plan      3333 non-null   object 
 5   voice mail plan         3333 non-null   object 
 6   number vmail messages   3333 non-null   int64  
 7   total day minutes       3333 non-null   float64
 8   total day calls         3333 non-null   int64  
 9   total day charge        3333 non-null   float64
 10  total eve minutes       3333 non-null   float64
 11  total eve calls         3333 non-null   int64  
 12  total eve charge        3333 non-null   float64
 13  total night minutes     3333 non-null   float64
 14  total night calls       3333 non-null   

In [44]:
df.columns

Index(['state', 'account length', 'area code', 'phone number',
       'international plan', 'voice mail plan', 'number vmail messages',
       'total day minutes', 'total day calls', 'total day charge',
       'total eve minutes', 'total eve calls', 'total eve charge',
       'total night minutes', 'total night calls', 'total night charge',
       'total intl minutes', 'total intl calls', 'total intl charge',
       'customer service calls', 'churn'],
      dtype='object')

The dataset (dt) was provided making it an existing data. It has 3333 rows, and 21 columns with data types ; **booleans(bool), integers(int64), strings(object), and float(float64)**.

## DATA PREPARATION

In [45]:
# Checking for duplicates
def duplicated(x):
    duplicates = []
    duplicates.append(x.duplicated().sum())
    for dups in duplicates:
        if dups >= 1 :
            print(f'There are duplicates')
        elif dups == 0:
            print(f'No duplicates')
        else:
            print(f'Error')
        
duplicated(df)


No duplicates


In [46]:
# Checking for Null values
def identify_missing_values(x):
    missing = []
    for i in df.isnull().any():
        missing.append(i)
    missing_set = set(missing)
    if (len(missing_set) == 1):
        output = print("The Data has no missing values")
    else:
        output = print("The Data has missing values.")

    return output


identify_missing_values(df)

The Data has no missing values


In [48]:
# Checking for the unique values in the Dataset.
df.nunique()

state                       51
account length             212
area code                    3
phone number              3333
international plan           2
voice mail plan              2
number vmail messages       46
total day minutes         1667
total day calls            119
total day charge          1667
total eve minutes         1611
total eve calls            123
total eve charge          1440
total night minutes       1591
total night calls          120
total night charge         933
total intl minutes         162
total intl calls            21
total intl charge          162
customer service calls      10
churn                        2
dtype: int64

The dataset has no duplicates or missing values. However, the phone number column is in string formart instead of integers. Therefore, since it is not going to affect the analysis, the column will be dropped.

## MODELING

## EVALUATION

## DEPLOYMENT